File: 2020/11/2020_11.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
int solveP1();
int solveP2();
int main()
{

    cout << "Part 1 Answer: " << solveP1() << endl;
    cout << "Part 2 Answer: " << solveP2() << endl;
    return 0;
}

int solveP1()
{
    ifstream f("11.txt");
    vector<string> arr;
    string x;
    getline(f, x);
    string spacer(x.size() + 2, '*');
    arr.push_back(spacer);
    arr.push_back('*' + x + '*');
    while (getline(f, x))
        arr.push_back('*' + x + '*');
    arr.push_back(spacer);
    f.close();
    for (int i = 0; i < arr.size(); i++)
        cout
            << arr[i] << endl;
    vector<string> temp = arr;
    do
    {
        temp = arr;
        for (int i = 1; i < arr.size() - 1; i++)
        {
            for (int j = 1; j < arr[i].size() - 1; j++)
            {
                int adjacent = 0;
                if (arr[i][j] == 'L')
                {
                    arr[i][j + 1] == '#' ? adjacent++ : adjacent;
                    arr[i][j - 1] == '#' ? adjacent++ : adjacent;
                    arr[i + 1][j] == '#' ? adjacent++ : adjacent;
                    arr[i - 1][j] == '#' ? adjacent++ : adjacent;
                    arr[i + 1][j + 1] == '#' ? adjacent++ : adjacent;
                    arr[i + 1][j - 1] == '#' ? adjacent++ : adjacent;
                    arr[i - 1][j + 1] == '#' ? adjacent++ : adjacent;
                    arr[i - 1][j - 1] == '#' ? adjacent++ : adjacent;
                    adjacent > 0 ? temp[i][j] == '#' : 0;
                }
                else if (arr[i][j] == '#')
                {
                    arr[i][j + 1] == '#' ? adjacent++ : adjacent;
                    arr[i][j - 1] == '#' ? adjacent++ : adjacent;
                    arr[i + 1][j] == '#' ? adjacent++ : adjacent;
                    arr[i - 1][j] == '#' ? adjacent++ : adjacent;
                    arr[i + 1][j + 1] == '#' ? adjacent++ : adjacent;
                    arr[i + 1][j - 1] == '#' ? adjacent++ : adjacent;
                    arr[i - 1][j + 1] == '#' ? adjacent++ : adjacent;
                    arr[i - 1][j - 1] == '#' ? adjacent++ : adjacent;
                    adjacent >= 4 ? temp[i][j] == 'L' : 0;
                }
            }
        }
        if (equal(arr.begin(), arr.end(), temp.begin()))
            break;
        arr = temp;
    } while (true);
    for (int i = 0; i < arr.size(); i++)
        cout
            << arr[i] << endl;
    return -1;
}
int solveP2()
{
    return -1;
}

File: 2020/11/2020_11.py
# Part 1

def p1():
    seats = []
    with open("11.txt") as f:
        seats = f.readlines()
    for i in range(len(seats)):
        seats[i] = "-"+seats[i].strip()+"-"
    seats.insert(0, "-"*len(seats[0]))
    seats.append("-"*len(seats[0]))

    while(True):
        temp = seats.copy()
        for i in range(1, len(seats)-1):
            for j in range(1, len(seats[i])-1):
                if seats[i][j] == "L":
                    c = 0
                    c += 1 if seats[i][j+1] == "#" else 0
                    c += 1 if seats[i][j-1] == "#" else 0
                    c += 1 if seats[i+1][j+1] == "#" else 0
                    c += 1 if seats[i+1][j-1] == "#" else 0
                    c += 1 if seats[i-1][j+1] == "#" else 0
                    c += 1 if seats[i-1][j-1] == "#" else 0
                    c += 1 if seats[i+1][j] == "#" else 0
                    c += 1 if seats[i-1][j] == "#" else 0
                    if (c == 0):
                        temp[i] = temp[i][:j]+"#"+temp[i][j+1:]
                elif seats[i][j] == "#":
                    c = 0
                    c += 1 if seats[i][j+1] == "#" else 0
                    c += 1 if seats[i][j-1] == "#" else 0
                    c += 1 if seats[i+1][j+1] == "#" else 0
                    c += 1 if seats[i+1][j-1] == "#" else 0
                    c += 1 if seats[i-1][j+1] == "#" else 0
                    c += 1 if seats[i-1][j-1] == "#" else 0
                    c += 1 if seats[i+1][j] == "#" else 0
                    c += 1 if seats[i-1][j] == "#" else 0
                    if (c >= 4):
                        temp[i] = temp[i][:j]+"L"+temp[i][j+1:]
        if temp == seats:
            break
        seats = temp.copy()
    count = 0
    for l in seats:
        count += l.count("#")
    return count


if __name__ == "__main__":
    print("Part 1 Answer:", p1())
    # print("Part 2 Answer:", p2())
